File: src/main/java/org/tasks/voice/VoiceCommandActivity.java
package org.tasks.voice;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.widget.Toast;

import com.todoroo.astrid.service.TaskCreator;

import org.tasks.R;
import org.tasks.injection.ForApplication;
import org.tasks.injection.InjectingAppCompatActivity;

import javax.inject.Inject;

public class VoiceCommandActivity extends InjectingAppCompatActivity {

    @Inject TaskCreator taskCreator;
    @Inject @ForApplication Context context;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        Intent intent = getIntent();

        switch (intent.getAction()) {
            case "com.google.android.gm.action.AUTO_SEND":
                final String text = intent.getStringExtra(Intent.EXTRA_TEXT);
                taskCreator.basicQuickAddTask(text);
                Toast.makeText(context, getString(R.string.voice_command_added_task), Toast.LENGTH_LONG).show();
                finish();
        }
    }
}

File: app/src/main/java/org/tasks/voice/VoiceCommandActivity.kt
package org.tasks.voice

import android.content.Context
import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.lifecycleScope
import com.todoroo.astrid.service.TaskCreator
import dagger.hilt.android.AndroidEntryPoint
import dagger.hilt.android.qualifiers.ApplicationContext
import kotlinx.coroutines.launch
import org.tasks.R
import org.tasks.analytics.Firebase
import org.tasks.extensions.Context.toast
import javax.inject.Inject

@AndroidEntryPoint
class VoiceCommandActivity : AppCompatActivity() {
    @Inject lateinit var taskCreator: TaskCreator
    @Inject @ApplicationContext lateinit var context: Context
    @Inject lateinit var firebase: Firebase

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        if (intent.action == AUTO_SEND) {
            lifecycleScope.launch {
                intent.getStringExtra(Intent.EXTRA_TEXT)
                        ?.takeUnless { it.isBlank() }
                        ?.let {
                            taskCreator.basicQuickAddTask(it)
                            toast(R.string.voice_command_added_task)
                            firebase.addTask("note_to_self")
                        }
                finish()
            }
        } else {
            finish()
        }
    }

    companion object {
        private const val AUTO_SEND = "com.google.android.gm.action.AUTO_SEND"
    }
}